adding-spring-security/leaveapp-initial/src/test/java/com/jdriven/leaverequest/LeaveRequestControllerSpringBootWebEnvMockTest.java

```diff
@@ -37,87 +42,93 @@ void beforeEach() {
 		repository.clear();
 	}
 
-	@Test
-	void testRequest() throws Exception {
-		mockmvc.perform(post("/request/{employee}", "Alice")
-				// XXX Authenticate as Alice when making this request
-				.param("from", "2019-11-30")
-				.param("to", "2019-12-03"))
-				.andExpect(status().isAccepted())
-				.andExpect(content().contentType(MediaType.APPLICATION_JSON))
-				.andExpect(jsonPath("$.employee").value("Alice"))
-				.andExpect(jsonPath("$.status").value("PENDING"));
+	@Nested
+	class AuthorizeUser {
+
```

Comment:
Let's add empty `@BeforeEach` methods here, maybe even with a TODO hint on adding a MockBean & mock the decode. That way all implementations will look the same more or less.

Reply:
Fixed!